src/Socket/socket.ts

```diff
@@ -459,8 +459,15 @@ export const makeSocket = (config: SocketConfig) => {
 		end(new Boom(msg || 'Intentional Logout', { statusCode: DisconnectReason.loggedOut }))
 	}
 
-	const requestPairingCode = async (phoneNumber: string): Promise<string> => {
-		authState.creds.pairingCode = bytesToCrockford(randomBytes(5))
+	const requestPairingCode = async (phoneNumber: string, customPairingCode?: string): Promise<string> => {
+		if (customPairingCode) {
```

Comment:
To avoid this double if / else, we can:

```ts
  const pairingCode = customPairingCode ?? bytesToCrockford(randomBytes(5))

  if (customPairingCode && !/^\d{8}$/.test(pairingCode))) {
    throw new Error('Custom pairing code must be exactly 8 digits')
  }

  authState.creds.pairingCode = pairingCode
.....
```